buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ThirdPartyAuditTask.java

```diff
@@ -115,19 +124,36 @@ public File getJarExpandDir() {
         );
     }
 
-    public void setExcludes(String... classes) {
-        excludes.clear();
+    public void allowMissingClasses(String... classesOrPackages) {
+        for (String each : classesOrPackages) {
+            missingClassExcludes.add(each);
+        }
+    }
+
+    public void allowViolations(String... violatingClasses) {
+        for (String each : violatingClasses) {
+            violationsExcludes.add(each);
+        }
+    }
+
+    public void allowAllMissingClasses() {
+        allowAllMissingClasses = true;
```

Comment:
Instead of a boolean, I wonder if it would be better to use `null` as an indicator for all? Then have a special string like "*" that can be passed to the method taking package names (and it looks like a wildcard on package/classnames). This way there is only one dsl method for each type of ignore.

Reply:
I like the empty array version more so we just have `ignoreMissingClasses()`.
 `*` wouldn't be bad if we supported more pattern types.

Reply:
I think ignoreMissingClasses by itself is confusing personally, but I would be interested to see what others think. This isn't crucial for checking this in, but I would raise it for discussion after the holidays.